migrate.py

```diff
@@ -239,9 +240,9 @@ def _backup_next_snapshot_from_dataset(self, dataset_name, snapshots: List[Dict[
             return None
 
         parent_restic_snapshot_id = None
-        ancestors_in_restic = [ancestor for ancestor in snapshots_in_restic if ancestor["creation"] < snapshot["creation"]]
-        if len(ancestors_in_restic) > 0:
-            parent_restic_snapshot_id = ancestors_in_restic[-1]["id"]
+        if len(snapshots_in_restic) > 0:
+            nearest = sorted(snapshots_in_restic, key=lambda s: abs(s["creation"] - snapshot["creation"]))
```

Comment:
I'm not sure, but I'd assume using the parent is better than the child, even if the time difference to the parent might be larger.
But maybe it does not really matter a lot.

Reply:
Why is it better? In the current implementation, the parent typically doesn't exist, yet, because we iterate backwards through ZFS snapshots.